2022/Q2/artifacts/pos-model/PoS-model.md

```diff
@@ -69,35 +78,21 @@ total_voting_power[] in Epoch to VotingPower //map from epoch to voting_power
 ```go
 tx_become_validator(validator_address, consensus_key, staking_reward_address)
 {
-  //VP:https://github.com/anoma/anoma/blob/master/proof_of_stake/src/validation.rs#L421
-  //https://github.com/anoma/anoma/blob/master/proof_of_stake/src/validation.rs#L459
-  //https://github.com/anoma/anoma/blob/master/proof_of_stake/src/validation.rs#L471
-  //COMMENT: In the code they check not only at cur_epoch but at all epochs between cur_epoch and the offset. Double-check
-  //COMMENT: I do not see in the code where they check that the consensus key is not an old one (this is in the docs).
-  //COMMENT: todo: check there are no consensus_key changes schedule between cur_epoch and the offset. If this necessary though? should
-  //not be enough with the state checks?
-  //https://github.com/anoma/anoma/blob/master/proof_of_stake/src/validation.rs#L515
-  var pre_state = read_epoched_field(validators[validator_address].state, cur_epoch)
-  if (pre_state == ⊥ && validator_address != staking_reward_address) then
-    //reward_address is not in the docs/spec validator struct
+  //check that become_validator has not been called before for validator_address
+  var cur_state = read_epoched_field(validators[validator_address].state, cur_epoch+pipeline_length)
```

Comment:
Very minor. I would rename `cur_state`, since it's the state at `cur_epoch+pipeline_length` and not at `cur_epoch`, so I find it a bit misleading.
Maybe just `state`? or `pipeline_state`? or something along those lines?

Reply:
Sure